openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -255,11 +260,25 @@ def objfunc(self, dv_dict):
 
             # Get the objective function evaluations
             for name, obj in iteritems(self.get_objectives()):
-                func_dict[name] = obj
+                if nproc > 1:
+                    owner = system._owning_ranks[name]
```

Comment:
Funny you should mention that.  This code was taken from MAUD when I was trying to fix one of my distrib comp tests, but just now I commented all of the bcasts out and all of the tests still pass.